aladdin.sh

```diff
@@ -54,39 +56,43 @@ function get_plugin_dir() {
 
 # Check for cluster name aliases and alias them accordingly
 function check_cluster_alias() {
-    cluster_alias=$(jq -r ".cluster_aliases.$CLUSTER_CODE" "$ALADDIN_CONFIG_FILE")
+    cluster_alias=$(jq -r ".cluster_aliases[\"$CLUSTER_CODE\"]" "$ALADDIN_CONFIG_FILE")
     if [[ $cluster_alias != null ]]; then
         export CLUSTER_CODE=$cluster_alias
     fi
 }
 
 function check_and_handle_init() {
     # Check if we need to force initialization
-    local last_launched_file init_every current_time previous_run
-    last_launched_file="$HOME/.infra/last_checked_${NAMESPACE}_${CLUSTER_CODE}"
-    init_every=3600
-    current_time="$(date +'%s')"
-    # create last launched file if it doesn't exit
-    mkdir -p "$(dirname "$last_launched_file")" && touch "$last_launched_file"
-    previous_run="$(cat "$last_launched_file")"
-    if [[ "$current_time" -gt "$((${previous_run:-0}+init_every))" || "$previous_run" -gt "$current_time" ]]; then
+    if ! test -f "$HOME/.aladdin/infra/installed"; then
         INIT=true
     fi
-    # Handle initialization logic
-    if "$INIT"; then
+    if $INIT; then
+        clear_cache_file
         "$SCRIPT_DIR"/infra_k8s_check.sh --force
-        enter_minikube_env
-        minikube addons enable ingress > /dev/null
         readonly repo_login_command="$(jq -r '.aladdin.repo_login_command' "$ALADDIN_CONFIG_FILE")"
         if [[ "$repo_login_command" != "null" ]]; then
             eval "$repo_login_command"
         fi
-        local aladdin_image="$(jq -r '.aladdin.repo' "$ALADDIN_CONFIG_FILE"):$(jq -r '.aladdin.tag' "$ALADDIN_CONFIG_FILE")"
-        docker pull "$aladdin_image"
-        echo "$current_time" > "$last_launched_file"
     else
         "$SCRIPT_DIR"/infra_k8s_check.sh
-        enter_minikube_env
+    fi
+    if test -z "$(get_or_set_cache "${ALADDIN_IMAGE}")"; then
+        docker pull "$ALADDIN_IMAGE"
+        get_or_set_cache "${ALADDIN_IMAGE}" $(time_plus_offset 3600*24)
+    fi
+
+    # Use host docker if available
+    if docker info > /dev/null 2>/dev/null; then
+        case "$OSTYPE" in
+            darwin*) USING_HOST_DOCKER=true ;;
+        esac
+        if $USING_HOST_DOCKER; then
+            echo "Using host docker engine for aladdin"
+        fi
+    fi
```

Comment:
Maybe a scope creep, but the aladdin container can run on the host docker engine

Reply:
I've used it for a while and seems to work fine, but would rather get this released separately so I've removed it from this PR